src/openforms/prefill/contrib/haalcentraal/plugin.py

```diff
@@ -72,25 +72,35 @@ def _get_values_for_bsn(
 
         return values
 
-    @classmethod
-    def get_prefill_values(
-        cls, submission: Submission, attributes: Sequence[str]
-    ) -> dict[str, Any]:
+    def get_identifier_value(
+        self, submission: Submission, identifier_role: str
+    ) -> str | None:
+        if not submission.is_authenticated:
+            return
+
         if (
-            not submission.is_authenticated
-            or submission.auth_info.attribute != AuthAttribute.bsn
+            identifier_role == IdentifierRoles.main
+            and submission.auth_info.attribute == self.requires_auth
         ):
-            #  If there is no bsn we can't prefill any values so just return
-            logger.info("No BSN associated with submission, cannot prefill.")
+            return submission.auth_info.value
+
+        if identifier_role == IdentifierRoles.authorised_person:
+            return submission.auth_info.machtigen.get("value")
```

Comment:
should we rename `value` to `identifier`? I'm worried that additional data might need to be stored in the future and "value" is too generic of a name.

Reply:
I liked `value`, because in the `AuthInfo` model we have the fields `plugin`, `attribute`, `value`. So I thought it would make sense to use `value` also for the machtigen data. But I can change it if you think `identifier` is better

Reply:
auth info has value because it only makes sense in that context of the entire auth info (it's basically a description of identifier), but the machtigen data is not well structured so I'd prefer the additional clarity.